### DISCUSSION TITLE
Drop Table for Integration Tests

### DISCUSSION THREAD

**Discussion by parkernilson** [pocketbase/pocketbase#3675] [Q&A] [answered]
I am trying to write some integration tests and would like to either: 1) drop a table, then recreate it or 2) delete all records in a table, between each test to create isolation between tests. I tried doing this using the pb.collections.delete and pb.collections.create functions, but I realized that these will create migrations each time they are called. 

One idea that I have which may work is to query all of the records in the table and then delete each one individually, however I am afraid this will be slow if I ever want to perform a test that uses a lot of documents.

Is there any way that I can do what I am trying to do in the current version?

Thank you!

**Comment by ganigeorgiev** [comment, marked as answer]
If you mean dropping a table via the web APIs, then no, there is no such endpoint at the moment (_see https://github.com/pocketbase/pocketbase/issues/2238_).

You can try to:

- Disable the automigrate functionality when running the tests by starting the executable with `--automigrate=0`.

- OR Create the truncate endpoint on your own. Here is an example using the JS `pb_hooks`:

    ```js
    // pb_hooks/main.pb.js

    // Example client-side request:
    // 
    // ```js
    // await pb.admins.authWithPassword("test@example.com", "1234567890");
    // 
    // await pb.send("/truncate/yourCollection", { method: "DELETE" });
    // ```
    routerAdd("DELETE", "/truncate/:nameOrId", (c) => {
        const collection = $app.dao().findCollectionByNameOrId(c.pathParam("nameOrId"));

        $app.dao().db()
            .newQuery("DELETE FROM " + collection.name)
            .execute();
    }, $apis.requireAdminAuth());
    ```

**Reply by ganigeorgiev** [reply]
_An alternative approach for JS integration tests could be also to start/stop the executable with each test. To avoid one test interfering with the database changes of another, you can create a copy of your test `pb_data` and point to it with each PocketBase start. See also https://github.com/pocketbase/pocketbase/discussions/603#discussioncomment-3707629._
